src/pydata_sphinx_theme/short_link.py

```diff
@@ -0,0 +1,111 @@
+"""A custom Transform object to shorten github and gitlab links."""
+
+from typing import Iterator
+from urllib.parse import ParseResult, urlparse, urlunparse
+
+from docutils import nodes
+from docutils.nodes import Node
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from sphinx.util.nodes import NodeMatcher
+
+
+def traverse_or_findall(node: Node, condition: str, **kwargs) -> Iterator[Node]:
+    """Triage node.traverse (docutils <0.18.1) vs node.findall.
+
+    TODO: This check can be removed when the minimum supported docutils version
+    for numpydoc is docutils>=0.18.1.
+    """
+    return (
+        node.findall(condition, **kwargs)
+        if hasattr(node, "findall")
+        else node.traverse(condition, **kwargs)
+    )
```

Comment:
IMO this would make more sense in `.utils` (or perhaps `.fixes` or `.shims`?)  But since I think it's only used this one place (right?) I guess let's leave it alone.

Reply:
@12rambau this is the one I want your second opinion about, whether it's worth it to move it to `.utils` (or some other file) now.

Reply:
it's true it could be reused, let me move it to `.utils`